posthog/hogql_queries/insights/trends/breakdown_values.py

```diff
@@ -0,0 +1,104 @@
+        event_name: str,
+        breakdown_field: str,
+        query_date_range: QueryDateRange,
+        histogram_bin_count: Optional[float] = None,
+    ):
+        self.team = team
+        self.event_name = event_name
+        self.breakdown_field = breakdown_field
+        self.query_date_range = query_date_range
+        self.histogram_bin_count = int(histogram_bin_count) if histogram_bin_count is not None else None
+
+    def get_breakdown_values(self) -> List[str]:
+        select_field = ast.Alias(alias="value", expr=ast.Field(chain=["properties", self.breakdown_field]))
+
+        query = parse_select(
+            """
+                SELECT groupArray(value) FROM (
+                    SELECT
+                        {select_field},
+                        count(*) as count
+                    FROM
+                        events e
+                    WHERE
+                        {events_where}
+                    GROUP BY
+                        value
+                    ORDER BY
+                        count DESC,
+                        value DESC
+                )
+            """,
+            placeholders={
+                "events_where": self._where_filter(),
+                "select_field": select_field,
+            },
+        )
+
+        if self.histogram_bin_count is not None:
+            expr = self._to_bucketing_expression()
+            query.select = [expr]
+
+        response = execute_hogql_query(
+            query_type="TrendsQueryBreakdownValues",
+            query=query,
+            team=self.team,
+        )
+
+        values = response.results[0][0]
+        return values
+
+    def _where_filter(self) -> ast.Expr:
+        filters: List[ast.Expr] = []
+
+        filters.append(parse_expr("notEmpty(e.person_id)"))
+        filters.extend(
+            [
+                parse_expr(
+                    "toTimeZone(timestamp, 'UTC') >= {date_from}",
+                    placeholders=self.query_date_range.to_placeholders(),
+                ),
+                parse_expr(
+                    "toTimeZone(timestamp, 'UTC') <= {date_to}",
+                    placeholders=self.query_date_range.to_placeholders(),
+                ),
+            ]
+        )
+
+        if self.event_name is not None:
+            filters.append(parse_expr("event = {event}", placeholders={"event": ast.Constant(value=self.event_name)}))
+
+        return ast.And(exprs=filters)
+
+    def _to_bucketing_expression(self) -> ast.Expr:
+        assert isinstance(self.histogram_bin_count, int)
+
+        if self.histogram_bin_count <= 1:
+            qunatile_expression = "quantiles(0,1)(value)"
+        else:
+            quantiles = []
+            bin_size = 1.0 / self.histogram_bin_count
+            for i in range(self.histogram_bin_count + 1):
+                quantiles.append(i * bin_size)
+
+            qunatile_expression = f"quantiles({','.join([f'{quantile:.2f}' for quantile in quantiles])})(value)"
```

Comment:
Is this `.2f` precision enough for a large bin count?

Reply:
Taken from the current implementation, seems like it's been good enough in the past 🤷